networkit/cpp/centrality/test/CentralityGTest.cpp

```diff
@@ -1329,6 +1358,30 @@ TEST_P(CentralityGTest, testTopHarmonicCloseness) {
     }
 }
 
+TEST_F(CentralityGTest, testTopHarmonicClosenessWithNodeList) {
+    METISGraphReader reader;
+    Graph G = reader.read("input/lesmis.graph");
+    constexpr count k = 10;
+    const std::vector<node> nodeList{0, 1, 2, 3, 4, 5, 6, 11, 27, 48};
+
+    // We expect complete TopHarmonicCloseness to not contain nodes 0-6,
+    // while restricted should. {11, 27, 48} should be present in both
+    // results (but position might differ).
+    for (bool useNBbound : {true, false}) {
```

Comment:
I'd recommend having two tests, one per heuristic. If this test fails, we'll not be able to tell which of the two heuristics was running.

Reply:
Not sure whether we want to also do this for `TopCloseness` (2x4 more tests with a lot of duplication).

Reply:
Yes, this would introduce duplication. One option to avoid it is to implement TopCloseness in separate parametric GTests where NBbound as another parameter of the tests.

Reply:
Yes, makes sense. My first idea was to split Centrality and Top/TopHarmonicCloseness tests into different files in order to avoid multiple `TEST_SUITE` definitions in `CentralityGTests.cpp`. I also saw the possibiliy to just add a list of parameters. However for bools this would not really improve the situation if something fails (without taking a dive into the code). Do you have other suggestions about how to implement this?

Reply:
Opted now for splitting the tests in different files. Having multiple test suites in one huge file decreases readability too much imho.